internal/emscripten/emscripten.go

```diff
@@ -89,8 +110,61 @@ func (v *InvokeFunc) Call(ctx context.Context, mod api.Module, stack []uint64) {
 		panic(err)
 	}
 
-	err = f.CallWithStack(ctx, stack)
+	// The Go implementation below mimics the Emscripten JS behaviour to support
+	// longjmps from indirect function calls. The implementation of these
+	// indirection function calls in Emscripten JS is like this:
+	//
+	// function invoke_iii(index,a1,a2) {
+	//  var sp = stackSave();
+	//  try {
+	//    return getWasmTableEntry(index)(a1,a2);
+	//  } catch(e) {
+	//    stackRestore(sp);
+	//    if (e !== e+0) throw e;
+	//    _setThrew(1, 0);
+	//  }
+	//}
+
+	// This is the equivalent of "var sp = stackSave();".
+	// We reuse savedStack to save allocations.
+	var savedStack [2]uint64
```

Comment:
maybe add a comment of reference to why `sp = stackSave` becomes two variables? like a source code link in emscripten?

Reply:
I have added a comment, it's not 2 variables for `stackSave()`, but it's 2 so that we can use it for `stackSave`, `stackRestore` and `setThrew`, `setThrew` needs 2 because it has 1 input and 1 output variable.

Reply:
aha makes sense